src/Lib.P2P.Tests/Protocols/VersionedNameTest.cs

```diff
@@ -0,0 +1,130 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using Lib.P2P.Protocols;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Lib.P2P.Tests.Protocols
+{
+    [TestClass]
+    public class VersionedNameTest
+    {
+        [TestMethod]
+        public void Parsing()
+        {
+            var vn = VersionedName.Parse("/multistream/1.0.0");
+            Assert.AreEqual("multistream", vn.Name);
+            Assert.AreEqual("1.0.0", vn.Version.ToString());
+
+            vn = VersionedName.Parse("/ipfs/id/1.0.0");
+            Assert.AreEqual("ipfs/id", vn.Name);
+            Assert.AreEqual("1.0.0", vn.Version.ToString());
+        }
+
+        [TestMethod]
+        public void Stringing()
+        {
+            var vn = new VersionedName {Name = "x", Version = new Semver.SemVersion(0, 42)};
+            Assert.AreEqual("/x/0.42.0", vn.ToString());
+        }
+
+        [TestMethod]
+        public void Value_Equality()
+        {
+            var a0 = VersionedName.Parse("/x/1.0.0");
+            var a1 = VersionedName.Parse("/x/1.0.0");
+            var b = VersionedName.Parse("/x/1.1.0");
+            VersionedName c = null;
+            VersionedName d = null;
+
+            Assert.IsTrue(c == d);
+            Assert.IsFalse(c == b);
+            Assert.IsFalse(b == c);
+
+            Assert.IsFalse(c != d);
+            Assert.IsTrue(c != b);
+            Assert.IsTrue(b != c);
+
+#pragma warning disable 1718
+            Assert.IsTrue(a0 == a0);
+            Assert.IsTrue(a0 == a1);
+            Assert.IsFalse(a0 == b);
+
+#pragma warning disable 1718
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not suppress issues.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)